### PR TITLE
[BFLY-1123] Biconomy

### PR DESCRIPTION
**Pull request by talhaah5**

### CODE REVIEW

File: src/exchange/biconomyService.ts [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+            const { data } = await perpContract.populateTransaction.addMargin(
+                getPublicAddress(),
+                toBigNumberStr(amount)
+            );
+            
+            txParams = {
+                data: data,
+                to: perpContract.address,
+                from: getPublicAddress(),
+                signatureType: "PERSONAL_SIGN"
```
Comment (malikkulsoom) [inline src/exchange/biconomyService.ts:55]: This should be enum

File: src/fireflyClient.ts [hunk 2]
```diff
@@ -188,6 +205,61 @@ export class FireflyClient {
...
+      } else {
+        provider = new HDWalletProvider({
+          privateKeys: [(this.getWallet() as Wallet).privateKey],
+          providerOrUrl: this.network.url,
+        });
+      }
+
+      this.marketSymbols = (await this.getMarketSymbols()).response.data;
+
+      this.marketSymbols = ['BTC-PERP']
```
Comment (malikkulsoom) [inline src/fireflyClient.ts:223]: please remove this

File: tests/fireflyClient.test.ts [hunk 3]
```diff
@@ -27,8 +27,8 @@ import {
 chai.use(chaiAsPromised);
 
 const testAcctKey =
-  "4d6c9531e0042cc8f7cf13d8c3cf77bfe239a8fed95e198d498ee1ec0b1a7e83";
-const testAcctPubAddr = "0xFEa83f912CF21d884CDfb66640CfAB6029D940aF";
+  "2e792a159afff0d33b9efb66c90f3b3905fcf18e649fdfa41696faba11eea154";
```
Comment (malikkulsoom) [inline tests/fireflyClient.test.ts:30]: please keep the original pvt and public key

File: tests/fireflyClient.test.ts [hunk 4]
```diff
@@ -41,7 +41,7 @@ describe("FireflyClient", () => {
   let sellPrice = 20000;
 
   before(async () => {
-    client = new FireflyClient(true, network, testAcctKey);
+    client = new FireflyClient(true, network, testAcctKey, false);
```
Comment (malikkulsoom) [inline tests/fireflyClient.test.ts:44]: uncomment this

File: tests/fireflyClient.test.ts [hunk 5]
```diff
@@ -72,12 +72,12 @@ describe("FireflyClient", () => {
   });
 
   beforeEach(async () => {
-    client = new FireflyClient(true, network, testAcctKey);
-    await client.init();
+    // client = new FireflyClient(true, network, testAcctKey);
```
Comment (malikkulsoom) [inline tests/fireflyClient.test.ts:75]: uncomment this

File: tests/fireflyClient.test.ts [hunk 6]
```diff
@@ -72,12 +72,12 @@ describe("FireflyClient", () => {
...
+    // await client.init();
   });
 
   afterEach(() => {
-    client.sockets.close();
+    // client.sockets.close();
```
Comment (malikkulsoom) [inline tests/fireflyClient.test.ts:80]: uncomment this

**Review by malikkulsoom — APPROVED** [review]

**Review by YameenMalik — APPROVED** [review]

**Pull request merged**